dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/monitor/DatadogRumMonitor.kt

```diff
@@ -14,15 +14,20 @@ import com.datadog.android.rum.RumMonitor
 import com.datadog.android.rum.RumResourceKind
 import com.datadog.android.rum.internal.domain.RumEvent
 import com.datadog.android.rum.internal.domain.RumEventData
+import com.datadog.android.rum.internal.domain.RumEventSerializer
+import java.lang.ref.WeakReference
 import java.util.UUID
 
 internal class DatadogRumMonitor(
     private val timeProvider: TimeProvider,
     private val writer: Writer<RumEvent>
 ) : RumMonitor {
 
-    private val activeView = mutableMapOf<Any, RumEvent>()
-    private val activeResources = mutableMapOf<Any, RumEvent>()
+    private var activeViewEvent: RumEvent? = null
```

Comment:
you should make all these fields volatile, we are not sure that all these calls will be made from the MainThread